Added Wolf Running Sound

Addehman opened the pull request.

Commit:
Added Wolf Running Sound

Commit:
Tuned the sounds' volumes and pitches, running sound now depends on velocity of player

MrJackey commented on Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -262,6 +264,24 @@ public class PlayerController : MonoBehaviour {
 			}
 		}
+
+		if (isGrounded && transformation.State == TransformationState.Wolf && Mathf.Abs(velocity.x) > 0.1f) {
+			if (!wolfRunningSound.isPlaying) {
+				wolfRunningSound.Play();
+			}
+		}
+		else {
+			wolfRunningSound.Stop();
+		}
+
+		if (isGrounded && transformation.State == TransformationState.Human && Mathf.Abs(velocity.x) > 0.1f) {
+			if (!humanRunningSound.isPlaying) {
+				humanRunningSound.Play();
+			}
+		}
+		else {
+			humanRunningSound.Stop();
+		}
```
If you start transforming the sound doesn't stop. This is due to returning the function above.

MrJackey replied:
This sound doesn't stop when you start crouching. It probably should.

MrJackey commented on Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -262,6 +264,24 @@ public class PlayerController : MonoBehaviour {
 			}
 		}
+
+		if (isGrounded && transformation.State == TransformationState.Wolf && Mathf.Abs(velocity.x) > 0.1f) {
+			if (!wolfRunningSound.isPlaying) {
+				wolfRunningSound.Play();
+			}
+		}
+		else {
+			wolfRunningSound.Stop();
+		}
+
+		if (isGrounded && transformation.State == TransformationState.Human && Mathf.Abs(velocity.x) > 0.1f) {
+			if (!humanRunningSound.isPlaying) {
+				humanRunningSound.Play();
+			}
+		}
+		else {
+			humanRunningSound.Stop();
+		}
```
Perhaps using xInput is better to use. Since velocity didn't really fix the problem of turning fast. Maybe using resume/pause instead of play/stop would solve this alongside falling down a small ledge.

Addehman replied:
applied it - it isn't perfect, but definitively better.. though I am reconsidering simply having it Play/Stop with xInput, might be weird if spammed, but otherwise felt best..

smeas replied:
You're using `Play`, not `Resume` in your last commit (10588d9). Play restarts the sound from the beginning.

Addehman replied:
But there is no "Resume" ?

smeas replied:
Oh, my bad! It's called `UnPause`. Also, you might have to call `Play` atleast once before using it. Not sure.

MrJackey reviewed (changes requested):
The wolf running sound hurts my ears, maybe skip/tune it or try to find something else. Other than that there is a few things, take a look at them.

Commit:
Applied suggested changes

lowered the volume and bass of the running sound for the wolf, this was amplified on my headphones, which are normally set pretty low, so didn't hear until I set them a bit higher! We'll see if this works otherwise we'll have to look for something else..

Also, I readjusted the timing of the steps to be a better fit with the wolf's running animation, please note this.

Commit:
Update PlayerController.cs

MrJackey commented on Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -215,6 +217,34 @@ public class PlayerController : MonoBehaviour {
+
+		if (isGrounded && !transformation.IsTransforming && 
+			transformation.State == TransformationState.Wolf && Mathf.Abs(xInput) > 0.1f) {
+			if (!wolfRunningSound.isPlaying && wolfRunningSound.time == 0) {
+				wolfRunningSound.Play();
+			}
+			else {
+				wolfRunningSound.UnPause();
+			}
```
This seems to be able to be simplified if you set the sound to play on Awake and pause it instantly in Start. 
```suggestion
			wolfRunningSound.UnPause();
```

MrJackey commented on Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -215,6 +217,34 @@ public class PlayerController : MonoBehaviour {
+
+		if (!isCrouched && !isCrouching && isGrounded && !transformation.IsTransforming &&
+			transformation.State == TransformationState.Human && Mathf.Abs(xInput) > 0.1f) {
+			if (!humanRunningSound.isPlaying && humanRunningSound.time == 0) {
+				humanRunningSound.Play();
+			}
+			else {
+				humanRunningSound.UnPause();
+			}
```
Same as above with wolfRunningSound
```suggestion
			humanRunningSound.UnPause();
```

MrJackey commented on Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -259,7 +289,7 @@ public class PlayerController : MonoBehaviour {
 			}
 			else {
-				crawlingSound.Stop();
+				crawlingSound.Pause();
```
This can probably be simplified like the other sounds. You should also move this sound handling above the return on 249 since the sound doesn't stop on transformation start.

MrJackey reviewed (changes requested):
A few things that I think can be changed, take a look. The new wolf running sound is much more pleasant for my ears 👍

Commit:
applied suggested changes

modified and simplified the crawl handling as well, please have a look!

MrJackey commented on Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -215,6 +221,30 @@ public class PlayerController : MonoBehaviour {
 				UpdateCrouchColliders(crouchCollider, humanCollider, crouchGroundCollider, humanGroundCollider);
 		}
+		
+		if (isCrouched && Mathf.Abs(xInput) > 0.1f) {
+					crawlingSound.UnPause();
+		}
+		else {
+			crawlingSound.Pause();
+		}
+
+		if (isGrounded && !transformation.IsTransforming && 
+			transformation.State == TransformationState.Wolf && Mathf.Abs(xInput) > 0.1f) {
+			wolfRunningSound.UnPause();
+		}
+		else {
+			wolfRunningSound.Pause();
+		}
+
+		if (!isCrouched && !isCrouching && isGrounded && !transformation.IsTransforming &&
+			transformation.State == TransformationState.Human && Mathf.Abs(xInput) > 0.1f) {
+			humanRunningSound.UnPause();
+		}
+		else {
+			humanRunningSound.Pause();
+		}
+
```
Could this be simplified to this?
```suggestion
if (Mathf.Abs(xInput) > 0.1f && isGrounded) {
			if (isCrouched) {
				crawlingSound.UnPause();
				humanRunningSound.Pause();
			}
			else if (transformation.State == TransformationState.Wolf) {
				wolfRunningSound.UnPause();
			}
			else if (transformation.State == TransformationState.Human && !isCrouching) {
				humanRunningSound.UnPause();
			}
		}
		else {
			crawlingSound.Pause();
			wolfRunningSound.Pause();
			humanRunningSound.Pause();
		}
```

Addehman replied:
Nice, much more efficient! :D

Commit:
Applied suggested changes

MrJackey reviewed (approved):
+1

smeas committed:
Merge branch 'master' into feature/211-wolf-running-steps

smeas reviewed (approved):
<img src="https://user-images.githubusercontent.com/8546813/101018772-44168a00-356c-11eb-9fea-3f6cf538d7f5.png" height="120">
